packages/http-server-js/eng/scripts/emit-e2e.js

```diff
@@ -0,0 +1,314 @@
+const ignoreFilePath = join(projectRoot, ".testignore");
+const logDirRoot = join(projectRoot, "temp", "emit-e2e-logs");
+const reportFilePath = join(logDirRoot, "report.txt");
+
+// Remove the log directory if it exists.
+async function clearLogDirectory() {
+  if (await pathExists(logDirRoot)) {
+    await rm(logDirRoot, { recursive: true, force: true });
+  }
+}
+
+// Parse command-line arguments.
+const argv = yargs(hideBin(process.argv))
+  .option("interactive", {
+    type: "boolean",
+    describe: "Enable interactive mode",
+    default: false,
+  })
+  .positional("paths", {
+    describe: "Optional list of specific file or directory paths to process (relative to basePath)",
+    type: "string",
+    array: true,
+    default: [],
+  })
+  .option("build", {
+    type: "boolean",
+    describe: "Build the generated projects",
+    default: false,
+  })
+  .help().argv;
+
+// Read and parse the ignore file.
+async function getIgnoreList() {
+  try {
+    const content = await readFile(ignoreFilePath, "utf8");
+    return content
+      .split(/\r?\n/)
+      .filter((line) => line.trim() && !line.startsWith("#"))
+      .map((line) => line.trim());
+  } catch {
+    console.warn(pc.yellow("No ignore file found."));
+    return [];
+  }
+}
+
+// Recursively process paths (files or directories relative to basePath).
+async function processPaths(paths, ignoreList) {
+  const results = [];
+  for (const relativePath of paths) {
+    const fullPath = resolve(basePath, relativePath);
+
+    if (!(await pathExists(fullPath))) {
+      console.warn(pc.yellow(`Path not found: ${relativePath}`));
+      continue;
+    }
+
+    const stats = await stat(fullPath);
+    if (stats.isFile() && fullPath.endsWith("main.tsp")) {
+      if (ignoreList.some((ignore) => relativePath.startsWith(ignore))) continue;
+      results.push({ fullPath, relativePath });
+    } else if (stats.isDirectory()) {
+      const patterns = ["**/main.tsp"];
+      const discoveredPaths = await globby(patterns, { cwd: fullPath });
+      const validFiles = discoveredPaths
+        .map((p) => ({
+          fullPath: join(fullPath, p),
+          relativePath: join(relativePath, p),
+        }))
+        .filter((file) => !ignoreList.some((ignore) => file.relativePath.startsWith(ignore)));
+      results.push(...validFiles);
+    } else {
+      console.warn(pc.yellow(`Skipping unsupported path: ${relativePath}`));
+    }
+  }
+
+  // Deduplicate.
+  const filesByDir = new Map();
+  for (const file of results) {
+    const dir = dirname(file.relativePath);
+    const existing = filesByDir.get(dir);
+    if (!existing) {
+      filesByDir.set(dir, file);
+    }
+  }
+  return Array.from(filesByDir.values());
+}
+
+// Run a shell command silently.
+async function runCommand(command, args, options = {}) {
+  // Remove clutter by not printing anything; capture output by setting stdio to 'pipe'.
+  return await execa(command, args, {
```

Comment:
could maybe use the `run` from internal build  utils or we should just rely on `execa` everywhere maybe instead

Reply:
Switched to using run. Note: Had to pass `silent: true` but then got similar output as execa.